src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,51 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
-    private final StorageDao storageDao = new StorageDaoImpl();
+    private final StorageDao storageDao;
+
+    public RegistrationServiceImpl(StorageDao storageDao) {
+        this.storageDao = storageDao;
+    }
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+        storageDao.add(user);
+        return user;
+    }
+
+    private void validateUser(User user) {
+        if (user == null) {
+            throw new RegistrationFailedException("Failed to add new user! \n" +
+                    "User must not be null");
```

Comment:
The exception message should be informative. While the message is clear, consider adding the specific reason for the failure, such as 'User object is null', to make it easier for the user to understand the exact validation failure.